src/test/java/SeleniumTest.java

```diff
@@ -0,0 +1,88 @@
+        timeSection.click();
+        WebElement attendanceMenu =  driver.findElement(By.id("menu_attendance_Attendance"));
+        attendanceMenu.click();
+        WebElement punchInOut =  driver.findElement(By.id("menu_attendance_punchIn"));
+        punchInOut.click();
+        WebElement startDate = driver.findElement(By.id("currentDate"));
+        WebElement startTime = driver.findElement(By.id("currentTime"));
+        String punchInDateTime = String.format("%s %s:00 GMT 2", startDate.getText(), startTime.getText());
+        WebElement inTextArea = driver.findElement(By.xpath("//textarea[@id='note']"));
+        inTextArea.sendKeys(startMessage);
+        WebElement buttonPunchIn = driver.findElement(By.id("btnPunch"));
+        buttonPunchIn.click();
+        driver.manage().timeouts().implicitlyWait(3, TimeUnit.SECONDS);
+        WebElement endDate = driver.findElement(By.id("currentDate"));
+        WebElement endTime = driver.findElement(By.id("currentTime"));
+        String punchOutDateTime = String.format("%s %s:00 GMT 2", endDate.getText(), endTime.getText());
+        WebElement outTextArea = driver.findElement(By.xpath("//textarea[@id='note']"));
+        outTextArea.sendKeys(endMessage);
+        WebElement buttonPunchOut = driver.findElement(By.id("btnPunch"));
+        buttonPunchOut.click();
+        attendanceMenu =  driver.findElement(By.id("menu_attendance_Attendance"));
+        attendanceMenu.click();
+        WebElement myRecords =  driver.findElement(By.id("menu_attendance_viewMyAttendanceRecord"));
+        myRecords.click();
+        WebElement dateField =  driver.findElement(By.id("attendance_date"));
+        dateField.clear();
+        dateField.sendKeys(todayDate + Keys.ENTER);
+        driver.manage().timeouts().implicitlyWait(3, TimeUnit.SECONDS);
+        List<WebElement> tableRows = driver.findElements(By.xpath(
+                "//form[@id='employeeRecordsForm']/table/tbody/tr[@class='odd' or @class='even']"));
+        WebElement latestRecord = tableRows.get(tableRows.size()-1);
+        List<WebElement> latestRecordBody = latestRecord.findElements(By.xpath("./td"));
+        String recordStartDateTime = latestRecordBody.get(1).getText();
+        String recordPunchInNote = latestRecordBody.get(2).getText();
+        String recordEndDateTime = latestRecordBody.get(3).getText();
+        String recordPunchOutNote = latestRecordBody.get(4).getText();
+        Assert.assertEquals(recordStartDateTime, punchInDateTime, "record Punch In datetime is different then actual Punch In datetime!");
+        Assert.assertEquals(recordPunchInNote, startMessage, "record Punch In note is different then actual Punch In note!");
+        Assert.assertEquals(recordEndDateTime, punchOutDateTime, "record Punch Out datetime is different then actual Punch Out datetime!");
+        Assert.assertEquals(recordPunchOutNote, endMessage, "record Punch Out datetime is different then actual Punch Out datetime!");
```

Comment:
this assertion fails

Reply:
can you please explain more on this?(please check my comments in Telegram)